src/proper_statem.erl

```diff
@@ -462,6 +480,189 @@ move_shrinker(Seq, Par, I) ->
+       ?LAZY(commands(Mod, InitialState)),
+       {finalize_weights(Mod, Cmds, maps:new()), Cmds}).
+
+%% @private
+-spec next_commands_gen(mod_name()) -> next_fun().
+next_commands_gen(Mod) ->
+  fun ({Weights, _Cmds}, {_D, T}) ->
+      NewWeights = next_weights(Weights, T),
+      CmdsGen = next_gen(Mod, NewWeights),
+      ?SHRINK(
+         ?LET(Cmds, ?LAZY(CmdsGen),
+              {finalize_weights(Mod, Cmds, NewWeights), Cmds}),
+         [CmdsGen])
+  end.
+
+%% @private
+-spec next_commands_gen(mod_name(), symbolic_state()) -> next_fun().
+next_commands_gen(Mod, InitialState) ->
+  fun ({Weights, _Cmds}, {_D, T}) ->
+      NewWeights = next_weights(Weights, T),
+      CmdsGen = next_gen(Mod, NewWeights, InitialState),
+      ?SHRINK(
+         ?LET(Cmds, ?LAZY(CmdsGen),
+              {finalize_weights(Mod, Cmds, NewWeights), Cmds}),
+         [CmdsGen])
+  end.
+
+%% Produce the new random weights by adding or removing an integer
+%% value to the existing weights.
+
+%% @private
+-spec next_weights(weights(), proper_gen_next:temperature()) -> weights().
+next_weights(Weights, Temp) ->
+  Factor = round(Temp * 5),
+  Base = ?RESIZE_FACTOR,
+  Min = -Base - Factor,
+  Max = Base + Factor,
+  maps:map(fun (_Key, W) ->
+               Random = rand:uniform(1 + (Max - Min)) - (1 - Min),
+               NW = W + Random,
+               case NW >= 1 of
+                 true -> NW;
+                 false -> 1
+               end
+           end, Weights).
+
+%% Used to return the generator for the commands of the targeted
+%% generation.
+
+%% @private
+-spec next_gen(mod_name(), weights()) -> proper_types:type().
+next_gen(Mod, Weights) ->
+  ?LET(InitialState, ?LAZY(Mod:initial_state()),
+       ?SUCHTHAT(
+          Cmds,
+          ?LET(List,
+               ?SIZED(Size,
+                      proper_types:noshrink(
+                        next_gen(Size * ?RESIZE_FACTOR, Mod,
+                                 InitialState, 1, Weights))),
+               proper_types:shrink_list(List)),
+          is_valid(Mod, InitialState, Cmds, []))).
+
+%% Used to return the generator for the commands of the targeted
+%% generation with an initial state provided by the user.
+
+%% @private
+-spec next_gen(mod_name(), weights(), symbolic_state()) -> proper_types:type().
+next_gen(Mod, Weights, InitialState) ->
+  ?SUCHTHAT(Cmds,
+            ?LET(CmdTail,
+                 ?LET(List,
+                      ?SIZED(Size,
+                             proper_types:noshrink(
+                               next_gen(Size * ?RESIZE_FACTOR, Mod,
+                                        InitialState, 1, Weights))),
+                      proper_types:shrink_list(List)),
+                 [{init, InitialState} | CmdTail]),
+            is_valid(Mod, InitialState, Cmds, [])).
+
+-spec next_gen(proper_gen:size(), mod_name(), symbolic_state(), pos_integer(),
+               weights()) -> proper_types:type().
+next_gen(Size, Mod, State, Count, Weights) ->
```

Comment:
My previous comment about the various `next_gen` functions being extensions of the previous one and the Erlang argument convention is applicable here too.  It's better as:
```erlang
next_gen(Mod, Weights, State, Mod, Count) ->
```

Reply:
This was "inspired" by the similar `commands/4` function, already present in this module. That's why the `Size` was put as the first argument.

Reply:
Perhaps there a good reason to leave `Size` first, but
1. without a comment in that function, nobody is aware of this reason, and
2. this does not explain the placement of `Weights` last.

Reply:
As long as it's OK with you, I will update my code :)